x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/DataFrameAnalyticsManager.java

```diff
@@ -222,6 +223,13 @@ private void reindexDataframeAndStartAnalysis(DataFrameAnalyticsTask task, DataF
                     task.markAsCompleted();
                     return;
                 }
+
+                Exception reindexError = getReindexError(task.getParams().getId(), reindexResponse);
+                if (reindexError != null) {
+                    task.markAsFailed(reindexError);
+                    return;
+                }
+
                 task.setReindexingTaskId(null);
```

Comment:
I think since the reindex task is done, wouldn't we want to mark it as `null` even when it failed?

Reply:
Good point, will do.